sdk/azidentity/workload_identity.go

```diff
@@ -7,18 +7,31 @@
 package azidentity
 
 import (
+	"bytes"
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"errors"
+	"fmt"
+	"net/http"
 	"os"
+	"strings"
 	"sync"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/errorinfo"
 )
 
-const credNameWorkloadIdentity = "WorkloadIdentityCredential"
+const (
+	aksCAData                = "AZURE_KUBERNETES_CA_DATA"
+	aksCAFile                = "AZURE_KUBERNETES_CA_FILE"
+	aksSNIName               = "AZURE_KUBERNETES_SNI_NAME"
+	aksTokenEndpoint         = "AZURE_KUBERNETES_TOKEN_ENDPOINT"
```

Comment:
nit: the SDK changes are generic and are meant to be usable on any kube running on Azure, i.e. AKS, CAPZ, etc.

```suggestion
	azureCAData                = "AZURE_KUBERNETES_CA_DATA"
	azureCAFile                = "AZURE_KUBERNETES_CA_FILE"
	azureSNIName               = "AZURE_KUBERNETES_SNI_NAME"
	azureTokenEndpoint         = "AZURE_KUBERNETES_TOKEN_ENDPOINT"
```